### PR TITLE
Cleanup public function definitions for doxygen

### PR DESCRIPTION
**Pull request by gquintin**
- Cleanup all public function declaration (primary and only objective at the beginning)
- Fix all mistakes I could see (indentation, spelling, missing docs)
- Fix all doxygen warnings except the "Compound [...] is not documented."'s

### CODE REVIEW

File: include/boost/simd/function/exponent.hpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ namespace boost { namespace simd
       @snippet exponent.txt exponent
 
   **/
-  as_integer_t<IEEEValue> exponent(IEEEValue const& x);
+  as_integer_t<Value> exponent(exValue const& x);
```
Comment (jfalcou) [inline include/boost/simd/function/exponent.hpp]: Why changing this ?

File: include/boost/simd/function/genmask.hpp [hunk 2]
```diff
@@ -16,13 +16,11 @@ namespace boost { namespace simd
 {
   /*!
     @ingroup group-bitwise
-    This function object returns a mask of bits. All ones if the
-    input element is non @ref Zero else all zeros.
+    This function object returns a mask of bits: full of ones if the
```
Comment (jfalcou) [inline include/boost/simd/function/genmask.hpp]: sentence is broken

File: include/boost/simd/function/genmaskc.hpp [hunk 3]
```diff
@@ -17,14 +17,13 @@ namespace boost { namespace simd
 
   /*!
     @ingroup group-bitwise
-    This function object returns a mask of bits. All ones if the
-    input element is @ref Zero else all zeros.
-
-    @see genmask
-
+    This function object returns a mask of bits: full of ones if the
```
Comment (jfalcou) [inline include/boost/simd/function/genmaskc.hpp]: you just wrote somethign else in genmask

File: include/boost/simd/function/frac.hpp [hunk 4]
```diff
@@ -17,7 +17,7 @@ namespace boost { namespace simd
 
   /*!
     @ingroup group-ieee
-    This function object returns the fractional part of the input i.e. @c x-trunc(x).
+    This function object returns the fractional part of the input i.e. `x-trunc(x)`.
```
Comment (jfalcou) [inline include/boost/simd/function/frac.hpp]: why the change ?

File: include/boost/simd/function/cosd.hpp [hunk 5]
```diff
@@ -25,8 +25,8 @@ namespace boost { namespace simd
 
     @par Note
 
-     - The semantics of the function are similar to cos ones.
-     See cos for further details
+     - The semantics of the function are similar to @ref cos ones.
+     See @ref cos for further details
```
Comment (jfalcou) [inline include/boost/simd/function/cosd.hpp]: this is useless as the first ref linsk already

File: include/boost/simd/function/cospi.hpp [hunk 6]
```diff
@@ -45,7 +45,7 @@ namespace boost { namespace simd
       @snippet cospi.txt cospi
 
   **/
-  IEEEValue cospi(IEEEValue const& x);
+  Value cospi(Value const& x);
```
Comment (jfalcou) [inline include/boost/simd/function/cospi.hpp]: Are you sure ?

File: include/boost/simd/function/interleave_even.hpp [hunk 7]
```diff
@@ -16,10 +16,7 @@ namespace boost { namespace simd
 #if defined(DOXYGEN_ONLY)
   /*!
     @ingroup group-swar
-    Function object implementing interleave_even capabilities
-
-    Computes a vector from a combination of the two inputs.
-
+    This function object cComputes a vector from a combination of the two inputs.
```
Comment (jfalcou) [inline include/boost/simd/function/interleave_even.hpp]: "computes"

File: include/boost/simd/function/inrad.hpp [hunk 8]
```diff
@@ -17,7 +17,7 @@ namespace boost { namespace simd
 
   /*!
     @ingroup group-trigonometric
-    This function object returns the convert degree to radian.
+    This function object converts degree to radian.
```
Comment (jfalcou) [inline include/boost/simd/function/inrad.hpp]: "converts its argument in degree to radian"

File: include/boost/simd/function/interleave_first.hpp [hunk 9]
```diff
@@ -16,10 +16,7 @@ namespace boost { namespace simd
 #if defined(DOXYGEN_ONLY)
   /*!
     @ingroup group-swar
-    Function object implementing interleave_first capabilities
-
-    Computes a vector from a combination of the two inputs.
-
+    This function object cComputes a vector from a combination of the two inputs.
```
Comment (jfalcou) [inline include/boost/simd/function/interleave_first.hpp]: "computes"

File: include/boost/simd/function/interleave_odd.hpp [hunk 10]
```diff
@@ -14,10 +14,7 @@ namespace boost { namespace simd
 #if defined(DOXYGEN_ONLY)
   /**
     @ingroup group-swar
-    Function object implementing interleave_odd capabilities
-
-    Computes a vector from a combination of the two inputs.
-
+    This function object cComputes a vector from a combination of the two inputs.
```
Comment (jfalcou) [inline include/boost/simd/function/interleave_odd.hpp]: "computes"

File: include/boost/simd/function/interleave_second.hpp [hunk 11]
```diff
@@ -17,10 +17,7 @@ namespace boost { namespace simd
   /*!
 
     @ingroup group-swar
-    Function object implementing interleave_second capabilities
-
-    Computes a vector from a combination of the two inputs.
-
+    This function object cComputes a vector from a combination of the two inputs.
```
Comment (jfalcou) [inline include/boost/simd/function/interleave_second.hpp]: "computes"

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/boost/simd/function/exponent.hpp (hunk 1 above)
Comment (jtlap) [reply include/boost/simd/function/exponent.hpp]: ex is a typo, but exponent admits integral enties

File: include/boost/simd/function/genmask.hpp (hunk 2 above)
Comment (jtlap) [reply include/boost/simd/function/genmask.hpp]: corrected

File: include/boost/simd/function/genmaskc.hpp (hunk 3 above)
Comment (jtlap) [reply include/boost/simd/function/genmaskc.hpp]: corrected

File: include/boost/simd/function/interleave_first.hpp (hunk 9 above)
Comment (jtlap) [reply include/boost/simd/function/interleave_first.hpp]: done

File: include/boost/simd/function/interleave_odd.hpp (hunk 10 above)
Comment (jtlap) [reply include/boost/simd/function/interleave_odd.hpp]: done

File: include/boost/simd/function/interleave_second.hpp (hunk 11 above)
Comment (jtlap) [reply include/boost/simd/function/interleave_second.hpp]: done

File: include/boost/simd/function/unary_minus.hpp [hunk 12]
```diff
@@ -18,39 +18,33 @@ namespace boost { namespace simd
...
+    @warningbox{Take care that the infix notation in scalar integral mode encurs
+    a possible promotion of the type as C++ rules state.}
 
-    or
 
-    @code
-    auto r = -x;
-    @endcode
+    @par Decorators
+    - `saturated_` ensurs that `saturated_(unary_minus)(Valmin<T>())` is `Valmin<T>()`
```
Comment (jfalcou) [inline include/boost/simd/function/unary_minus.hpp]: "ensures"

**Review by jfalcou — APPROVED** [review]

**Pull request merged**